opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp

```diff
@@ -220,6 +219,11 @@ namespace Opm {
         // Numerical aquifer cells, needs to be active
         std::unordered_set<size_t> m_aquifer_cells;
 
+        // Radial grids need this for volume calculations.
+        bool m_is_cylindrical = false;
```

Comment:
Instead of using a separate bool for this you could use `std::optional<std::vector<double>>` for the datavectors.